### PR TITLE
Handle exceptions occurred via scheduler in main method

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
Fix the $Subject. Code will be generated as below.

```java

Scheduler var1 = new Scheduler(4);
FutureValue var2 = var1.schedule(new Object[1], ___init::$lambda$___init_$, (Strand)null);
var1.start();
if (var2.panic != null) {
    var2.panic.printStackTrace();
} else {
    FutureValue var3 = var1.schedule(new Object[1], ___init::$lambda$main$, (Strand)null);
    var3.strand.frames = new Object[100];
    var1.start();
    if (var3.panic != null) {
        var3.panic.printStackTrace();
    }
}

```

This also fixes the printing garbage values when instruction doesn't have positions in bir emitter.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BallerinaErrors.java [hunk 1]
```diff
@@ -72,76 +70,4 @@ static ErrorValue createNumericConversionError(Object inputValue, BType targetTy
                                      RuntimeErrors.INCOMPATIBLE_SIMPLE_TYPE_CONVERT_OPERATION,
                                      TypeChecker.getType(inputValue), inputValue, targetType));
     }
-
-    public static String getPrintableStackTrace(ErrorValue error) {
-        return getPrintableStackTrace(getErrorMessage(error), error.getStackTrace());
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BallerinaErrors.java]: Move this to ErrorValue class so we can override throwable printstacktrace method and use e.printstacktrace for ErrorValue as well.

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BallerinaErrors.java [hunk 2]
```diff
@@ -40,6 +35,9 @@
 
     public static final String NULL_REF_EXCEPTION = "NullReferenceException";
 
+    public static final String ERROR_PRINT_PREFIX = "error: ";
+
+
```
Comment (vinok88) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BallerinaErrors.java]: Extra line?

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal [hunk 3]
```diff
@@ -870,10 +870,7 @@ function generateMainMethod(bir:Function userMainFunc, jvm:ClassWriter cw, bir:P
 
     BalToJVMIndexMap indexMap = new;
     ErrorHandlerGenerator errorGen = new(mv, indexMap);
-    jvm:Label endLabel = new;
-    jvm:Label handlerLabel = new;
-    errorGen.generateTryIns(endLabel, handlerLabel);
-
+
```
Comment (vinok88) [inline compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal]: Remove extra line.

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal [hunk 4]
```diff
@@ -886,8 +883,10 @@ function generateMainMethod(bir:Function userMainFunc, jvm:ClassWriter cw, bir:P
     mv.visitInsn(DUP);
     mv.visitInsn(ICONST_4);
     mv.visitMethodInsn(INVOKESPECIAL, SCHEDULER, "<init>", "(I)V", false);
+    mv.visitVarInsn(ASTORE, 1);
```
Comment (vinok88) [inline compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal]: Can you hardcode the var index to 1 here?

**Comment by Kishanthan** [comment]
Related to #15370

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal (hunk 4 above)
Comment (warunalakshitha) [reply compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_method_gen.bal]: Fixed

**Pull request merged**